### DISCUSSION TITLE
Manually triggered dags are queued and never executed

### DISCUSSION THREAD

**Discussion by match-gabeflores** [apache/airflow#17837] [Q&A]
We set up a v2.1.3 instance and whenever we manually trigger a dag, it stays queued. It never starts running. 

However, all the scheduled dag executions _do_ start running and finish successfully. Do we have something misconfigured? I know there were a lot of changes to Task States and Queued states in this release. 

v2.1.3, Python 3.9, Postgres 12, Local Executor

![image](https://user-images.githubusercontent.com/286903/130872949-af421f8f-6b40-4db0-9580-1662f8f19b8b.png)

sample test dag below

```
from airflow import DAG
from airflow.operators.python import PythonOperator

from datetime import timedelta, datetime

########################################################################################################################
# Define custom code
########################################################################################################################


def move_files_in_s3():
    print ("hi")

########################################################################################################################
# Define DAG and setup tasks and workflow
########################################################################################################################
default_args = {
    'owner': 'TechAnalytics',
    'depends_on_past': False,
    'start_date': datetime(2020, 9, 1),
    'retries': 3,
    'retry_delay': timedelta(minutes=15)
}

dag = DAG(
    dag_id='test',
    default_args=default_args,
    description='Load Match page data to RS',
    schedule_interval='*/5 * * * *',  
    catchup=False
)

with dag:

    #####################################################################
    # OPERATORS
    # change this to python operator to copy and delete at same time
    move_incoming_to_working = PythonOperator(
        task_id="copy_incoming_to_working",
        python_callable=move_files_in_s3        
    )


    # set up workflow order
    move_incoming_to_working
````

**Comment by kaxil** [comment]
cc @ephraimbuddy our in-house expert in this.

**Reply by kaxil** [reply]
Do you manually trigger from the UI, API or CLI?

**Reply by kaxil** [reply]
Also can you post Scheduler logs please, if you don't see anything suspicious, please enable DEBUG logging: https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#logging-level

**Reply by match-gabeflores** [reply]
> Do you manually trigger from the UI, API or CLI?

From UI. From CLI, seems to work fine.

**Reply by ephraimbuddy** [reply]
I ran your dag on 2.1.3 and main and I was not able to reproduce your issue. I used SequentialExecutor and LocalExecutor. Could you share more reproduce steps

**Reply by ephraimbuddy** [reply]
![manualtrigger](https://user-images.githubusercontent.com/4122866/130918267-b0b5fd75-74a5-4ff0-b65b-4301c1535f9f.PNG)

**Reply by match-gabeflores** [reply]
v2.1.3, Python 3.9, Postgres 12, Local Executor. On a empty dev instance, just have two dags on there for testing. 

Posted on slack and someone else had the same issue. But looks like a VM restart worked for them. We didn't get a chance to try it out. 
https://apache-airflow.slack.com/archives/CCR6P6JRL/p1629947021001100?thread_ts=1629929832.351400&cid=CCR6P6JRL
![image](https://user-images.githubusercontent.com/286903/131003966-79eaaa78-3348-4c14-87bd-38bc4771ac70.png)


Our ops team downgraded to 2.1.1 (and python 3.8) and we aren't having any issues. (we're in a time crunch, unfortunately). Wish I could have provided more logs.

**Reply by argibbs** [reply]
FWIW I have seen similar behaviour, which I've noted in #17976. It affected manual but also (primarily) scheduled dags. It seemed related to when there were several dags with catchup=`True` and large backlogs to work through.

The problem seems fairly clearly related to the creation of dag runs with the explicit "queued" state, which is AFAIK a new thing in 2.1.3.

Like @match-gabeflores I am under the cosh a bit, and have had to drop back from 2.1.3 to 2.1.2, which fixed my problem instantly. I am a sucker for the bleeding edge, so will try again with 2.1.4 / 2.2 once available.

**Reply by ephraimbuddy** [reply]
@argibbs I think what you experienced is related to how `max_active_runs` work right now. I have added a fix for that here: https://github.com/apache/airflow/pull/17945. I also think limiting the number of 'queued' dagrun is necessary. Thanks for reporting

**Reply by argibbs** [reply]
Awesome, yeah I thought #17945 looked really promising. The minute there's a release out with it in, I'll give it a try. Ty!

**Reply by match-gabeflores** [reply]
@ephraimbuddy - Great to hear. We're also eager to have a working `max_active_runs ` implementation. Thanks for all the debugging!
